webpack/assets/javascripts/react_app/components/bookmarks/form/form.test.js

```diff
@@ -1,108 +1,65 @@
 import { Provider } from 'react-redux';
-import BookmarkForm from './';
-import { generateStore } from '../../../redux';
-import * as FormActions from '../../../redux/actions/common/forms';
+import nock from 'nock';
+import BookmarkFormContainer, { BookmarkForm } from './';
+import { setupIntegrationTest, flushAllPromises } from '../../../IntergrationHelper';
+import bookmarks from '../../../redux/reducers/bookmarks/';
 
 function setup() {
   const props = {
     controller: 'hosts',
-    url: '/api/bookmarks',
+    url: 'http://localhost/api/bookmarks',
     onCancel: jest.fn(),
   };
-
-  const wrapper = mount(<Provider store={generateStore()}>
-      <BookmarkForm {...props} />
-    </Provider>);
+  const mockSubmitForm = jest.fn();
+  const wrapper = shallow(<BookmarkForm {...props} handleSubmit={mockSubmitForm}/>);
 
   return {
     props,
     wrapper,
   };
 }
 
-describe('bookmark form', () => {
-  it('should include the correct initial values', () => {
-    const { wrapper } = setup();
+describe('bookmark form integration test', () => {
+  let store;
+  let dispatchSpy;
 
-    expect(wrapper.find('BookmarkForm').props().initialValues).toEqual({
-      publik: true,
-      query: '',
-    });
+  beforeEach(() => {
+    ({ store, dispatchSpy } = setupIntegrationTest({ bookmarks }));
   });
-  it('should render the entire form', () => {
-    const { wrapper } = setup();
-
-    expect(toJson(wrapper)).toMatchSnapshot();
-  });
-  it('should convert params for submittion correctly', () => {
-    const spy = jest.spyOn(FormActions, 'submitForm');
-    const { wrapper } = setup();
-
-    wrapper.find('input [name="name"]').simulate('change', { target: { value: 'Joe' } });
-    wrapper.find('textarea [name="query"]').simulate('change', { target: { value: 'search' } });
-    wrapper.find('input [name="publik"]').simulate('change', { target: { value: true } });
+  it('full flow', async () => {
+    nock('http://localhost')
+      .post('/api/bookmarks')
+      .reply(201, { ok: true, id: 70 });
+    const { props } = setup();
 
+    const wrapper = mount(<Provider store={store}>
+        <BookmarkFormContainer {...props} />
+      </Provider>);
+    wrapper
+      .find('input [name="name"]')
+      .simulate('change', { target: { value: 'Joe' } });
+    wrapper
+      .find('textarea [name="query"]')
+      .simulate('change', { target: { value: 'search' } });
+    wrapper
+      .find('input [name="publik"]')
+      .simulate('change', { target: { value: true } });
     wrapper.find('form').simulate('submit');
-    expect(wrapper.find('.spinner').length).toEqual(1);
-    expect(spy).toHaveBeenCalledWith({
-      item: 'Bookmark',
-      url: '/api/bookmarks',
-      values: {
-        controller: 'hosts', name: 'Joe', public: true, query: 'search',
-      },
+    await flushAllPromises();
+    expect(dispatchSpy).toHaveBeenCalledWith({
+      type: 'BOOKMARK_FORM_SUBMITTED',
+      payload: { item: 'Bookmark', body: { ok: true, id: 70 } },
```

Comment:
You can use snapshot here.
https://gist.github.com/sharvit/cf658a8494758be8062a63121b925fdc#file-bookmarksactions-test-js-L16